File: 0016-3sum-closest/0016-3sum-closest.py
import sys
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        result = sys.maxsize 
        for i in range(0, len(nums)-2):
            left = i+1
            right = len(nums) - 1
            while left < right:
                sums = nums[i] + nums[left] + nums[right]
                if(abs(target-sums) < abs(target-result)):
                    result = sums
                if sums < target:
                    left += 1
                else:
                    right -= 1
        return result
        
        return 1
        

File: 0016-3sum-closest/0016-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        Arrays.sort(nums);
        int result = Integer.MAX_VALUE;
        for(int i=0;i<nums.length-2;i++){
            int left = i+1;
            int right = nums.length-1;
            while(left < right){
                int sum = nums[i] + nums[left] + nums[right];
                if(Math.abs(target-sum) < Math.abs(target-result)) {
                    result = sum;
                }
                if(sum < target){
                    left ++;
                } else {
                    right --;
                }
            }
        }
        return result;
    }
}